File: solutions/fsharp/raindrops/Raindrops.fs
module Raindrops

let convert (number: int): string =
    let sound divider word =
        if number % divider = 0 then word else ""
    [ sound 3 "Pling"
      sound 5 "Plang"
      sound 7 "Plong" ]
    |> String.concat ""
    |> function
    | "" -> string number
    | sounds -> sounds

File: solutions/scala/raindrops/src/main/scala/Raindrops.scala
object Raindrops {
  def convert(n: Int): String =
    var sound = ""
    if n % 3 == 0 then sound += "Pling" 
    if n % 5 == 0 then sound += "Plang"
    if n % 7 == 0 then sound += "Plong"
    if sound.isEmpty then n.toString else sound
}